packages/fx-core/src/plugins/resource/bot/configs/teamsBotConfig.ts

```diff
@@ -23,60 +23,60 @@ export class TeamsBotConfig {
   public actRoles: PluginActRoles[] = [];
   public resourceNameSuffix = "";
 
-  public async restoreConfigFromContext(context: PluginContext): Promise<void> {
-    await this.scaffold.restoreConfigFromContext(context);
-    await this.provision.restoreConfigFromContext(context);
-    await this.localDebug.restoreConfigFromContext(context);
-    await this.deploy.restoreConfigFromContext(context);
+  private constructor() {
+    this.scaffold = new ScaffoldConfig();
+    this.provision = new ProvisionConfig();
+    this.localDebug = new LocalDebugConfig();
+    this.deploy = new DeployConfig();
+  }
+
+  public static async restoreConfigFromContext(context: PluginContext): Promise<TeamsBotConfig> {
+    let teamsBotConfig = new TeamsBotConfig();
+    await teamsBotConfig.scaffold.restoreConfigFromContext(context);
+    await teamsBotConfig.provision.restoreConfigFromContext(context);
+    await teamsBotConfig.localDebug.restoreConfigFromContext(context);
+    await teamsBotConfig.deploy.restoreConfigFromContext(context);
 
-    const clientIdValue: ConfigValue = context.configOfOtherPlugins
+    teamsBotConfig.teamsAppClientId = context.configOfOtherPlugins
```

Comment:
I believe the change is slightly huge, but the key point is to make TeamsBotConfig stateless, that is, make it as a complete mirror of the context since solution will pass a fresh context whenever a new project is created.

So keep this target in mind, one thing to do is:
change code like:
` if (context.value is not empty/null) config.value = context.value` 
to
` config.value = context.value`

Reply:
Extra changes are kind of coding style changes, it's not related too much to fix the bug.

Reply:
Also, ut cases need to be updated accordingly.

Reply:
resolved